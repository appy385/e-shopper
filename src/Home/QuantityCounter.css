.count {
    border: maroon solid 0.2px;
}