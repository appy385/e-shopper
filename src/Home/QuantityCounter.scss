
.quantity-container {
    background-color: #F6CF72;
    width:fit-content;
    border: 0.1rem solid 0.1rem;
    align-self: center;
}
.quantity-basket-count {
    border: none;
    background-color: #F6CF72;
    width: 2rem;
    height: 2rem;
    font-weight: 500;
    font-size: 0.9rem;
    padding: 0 0.1rem 0 0.1rem;
    cursor: pointer;
}

.quantity-dec-button {
    border-radius: 0.3rem 0.3rem 0.3rem 0.3rem;
}

.quantity-inc-button {
    border-radius: 0.3rem 0.3rem 0.3rem 0.3rem;
}


