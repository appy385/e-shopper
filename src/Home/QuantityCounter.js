import PropTypes from 'prop-types';

import React from 'react';

import './QuantityCounter.css';

function QuantityCounter({ count, onIncrement, onDecrement }) {
  return (
    <div className="quantity-container">
      <button type="button" className="quantity-inc-button" onClick={onIncrement}>+</button>
      <span className="quantity-basket-count">
        {`${count} in Basket`}
      </span>
      <button type="button" className="quantity-dec-button" onClick={onDecrement}>-</button>
    </div>
  );
}

QuantityCounter.propTypes = {
  count: PropTypes.number.isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,

};

export default QuantityCounter;
