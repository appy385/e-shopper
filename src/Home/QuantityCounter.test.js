import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import QuantityCounter from './QuantityCounter';

describe(QuantityCounter.name, () => {
  let mockCount;
  let mockInc;
  let mockDec;
  beforeEach(() => {
    mockCount = 0;
    mockInc = jest.fn();
    mockDec = jest.fn();
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  test('should match snapshot for Card component', () => {
    const { container } = render(<QuantityCounter
      count={mockCount}
      onIncrement={mockInc}
      onDecrement={mockDec}
    />);
    expect(container).toMatchSnapshot();
  });
  test('should display + and - button', () => {
    render(<QuantityCounter
      count={mockCount}
      onIncrement={mockInc}
      onDecrement={mockDec}
    />);
    const incrementCount = screen.getByText('+');
    const decrementCount = screen.getByText('-');
    expect(incrementCount.tagName).toBe('BUTTON');
    expect(decrementCount.tagName).toBe('BUTTON');
  });
  test('should call mockInc func when + buttton is clicked', async () => {
    render(<QuantityCounter
      count={mockCount}
      onIncrement={mockInc}
      onDecrement={mockDec}
    />);
    const incrementCount = screen.getByText('+');
    fireEvent.click(incrementCount);
    expect(mockInc).toHaveBeenCalled();
  });
  test('should call mockDec func when - buttton is clicked', async () => {
    render(<QuantityCounter
      count={mockCount}
      onIncrement={mockInc}
      onDecrement={mockDec}
    />);
    const decrementCount = screen.getByText('-');
    fireEvent.click(decrementCount);
    expect(mockDec).toHaveBeenCalled();
  });
  test('should display count of product', async () => {
    render(<QuantityCounter
      count={mockCount}
      onIncrement={mockInc}
      onDecrement={mockDec}
    />);
    screen.getByText(`${mockCount} in Basket`);
  });
});
