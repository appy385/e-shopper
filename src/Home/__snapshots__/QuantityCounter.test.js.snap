// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityCounter should match snapshot for Card component 1`] = `
<div>
  <div
    class="quantity-container"
  >
    <button
      class="quantity-inc-button"
      type="button"
    >
      +
    </button>
    <span
      class="quantity-basket-count"
    >
      0 in Basket
    </span>
    <button
      class="quantity-dec-button"
      type="button"
    >
      -
    </button>
  </div>
</div>
`;
